src/main/java/com/artipie/api/RepositoryRest.java

```diff
@@ -0,0 +1,182 @@
+package com.artipie.api;
+
+import com.artipie.misc.JavaResource;
+import com.artipie.settings.repo.CrudRepoSettings;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.FileSystemAccess;
+import io.vertx.ext.web.handler.StaticHandler;
+import io.vertx.ext.web.openapi.RouterBuilder;
+import java.io.StringReader;
+import javax.json.JsonObject;
+import javax.json.JsonStructure;
+import javax.json.spi.JsonProvider;
+import org.eclipse.jetty.http.HttpStatus;
+
+/**
+ * Rest-api operations for repositories settings CRUD
+ * (create/read/update/delete) operations.
+ * @since 0.26
+ */
+@SuppressWarnings({"PMD.OnlyOneReturn", "PMD.AvoidDuplicateLiterals"})
+public final class RepositoryRest extends BaseRest {
+    /**
+     * Repository settings create/read/update/delete.
+     */
+    private final CrudRepoSettings crs;
+
+    /**
+     * Artipie layout.
+     */
+    private final String layout;
+
+    /**
+     * Ctor.
+     * @param crs Repository settings create/read/update/delete
+     * @param layout Artipie layout
+     */
+    public RepositoryRest(final CrudRepoSettings crs, final String layout) {
+        this.crs = crs;
+        this.layout = layout;
+    }
+
+    @Override
+    public void init(final RouterBuilder rbr) {
+        rbr.operation("listAll")
+            .handler(this::listAll)
+            .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        if ("flat".equals(this.layout)) {
+            rbr.operation("getRepo")
+                .handler(this::getRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("createRepo")
+                .handler(this::createRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        } else {
+            rbr.operation("list")
+                .handler(this::listUserRepos)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("getUserRepo")
+                .handler(this::getRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("createUserRepo")
+                .handler(this::createRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        }
+        final Router router = rbr.createRouter();
+        router.route("/api/*").handler(
+            StaticHandler.create(
+                FileSystemAccess.ROOT,
+                new JavaResource("swagger-ui").uri().getPath()
+            )
+        );
+    }
+
+    /**
+     * Get a repository settings json.
+     * @param context Routing context
+     */
+    private void getRepo(final RoutingContext context) {
+        final RepositoryName rname = new RepositoryName.FromRequest(context, this.layout);
+        if (!this.crs.exists(rname)) {
```

Comment:
When repo name is a service name ("_storages", "_permissions", "_credentials"), we should give to user a proper error message. I think, NOT_FOUND is misleading in that case.

Reply:
Thanks for your commets, please see change-commit.